configreader/reader_test.go

```diff
@@ -0,0 +1,194 @@
+		},
+	})
+	staticBootstrap = &v3.Bootstrap{
+		Admin: &v3.Admin{
+			AccessLogPath: "/tmp/admin_access.log",
+			Address: &envoy_config_core_v3.Address{
+				Address: &envoy_config_core_v3.Address_SocketAddress{
+					SocketAddress: &envoy_config_core_v3.SocketAddress{
+						Address: "127.0.0.1",
+						PortSpecifier: &envoy_config_core_v3.SocketAddress_PortValue{
+							PortValue: 9901,
+						},
+					},
+				},
+			},
+		},
+		StaticResources: &v3.Bootstrap_StaticResources{
+			Clusters: []*v36.Cluster{
+				{
+					Name: "some_service",
+					ClusterDiscoveryType: &v36.Cluster_Type{
+						Type: v36.Cluster_STATIC,
+					},
+					ConnectTimeout: &duration.Duration{Nanos: 250000000},
+					LoadAssignment: &v31.ClusterLoadAssignment{
+						ClusterName: "some_service",
+						Endpoints: []*v31.LocalityLbEndpoints{
+							{
+								LbEndpoints: []*v31.LbEndpoint{
+									{
+										HostIdentifier: &envoy_config_endpoint_v3.LbEndpoint_Endpoint{
+											Endpoint: &envoy_config_endpoint_v3.Endpoint{
+												Address: &envoy_config_core_v3.Address{
+													Address: &envoy_config_core_v3.Address_SocketAddress{
+														SocketAddress: &envoy_config_core_v3.SocketAddress{
+															Address: "127.0.0.1",
+															PortSpecifier: &envoy_config_core_v3.SocketAddress_PortValue{
+																PortValue: 1234,
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			Listeners: []*v35.Listener{
+				{
+					Name: "listener_0",
+					Address: &envoy_config_core_v3.Address{
+						Address: &envoy_config_core_v3.Address_SocketAddress{
+							SocketAddress: &envoy_config_core_v3.SocketAddress{
+								Address: "127.0.0.1",
+								PortSpecifier: &envoy_config_core_v3.SocketAddress_PortValue{
+									PortValue: 10000,
+								},
+							},
+						},
+					},
+					FilterChains: []*envoy_config_listener_v3.FilterChain{
+						{
+							Filters: []*envoy_config_listener_v3.Filter{
+								{
+									Name: "envoy.filters.network.http_connection_manager",
+									ConfigType: &v35.Filter_TypedConfig{
+										TypedConfig: httpConnectionManager,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	if err != nil {
+		panic(err)
+	}
+}
+
+func TestReader(t *testing.T) {
+	type readerTest struct {
+		inputFile      string
+		expectedOutput *v3.Bootstrap
+	}
+
+	tests := map[string]readerTest{
+		"basic": {
+			inputFile:      "../testdata/static.yaml",
+			expectedOutput: staticBootstrap,
+		},
+	}
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			bootstrap, err := FromFile(test.inputFile)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			compareBootstrap(t, test.expectedOutput, bootstrap)
+		})
+	}
+}
+
+func compareBootstrap(t *testing.T, expected, actual *v3.Bootstrap) {
+	expectedBytes, err := protojson.Marshal(expected)
```

Comment:
Add `t.Helper()` here so the test framework points at where you used this func, not at the line in this func that had an error.